zk_prover/src/merkle_sum_tree/tree.rs

```diff
@@ -64,14 +142,49 @@ pub trait Tree<const N_ASSETS: usize, const N_BYTES: usize> {
     fn verify_proof(&self, proof: &MerkleProof<N_ASSETS, N_BYTES>) -> bool
     where
         [usize; N_ASSETS + 1]: Sized,
-        [usize; 2 * (1 + N_ASSETS)]: Sized,
+        [usize; N_ASSETS + 2]: Sized,
     {
-        let mut node = proof.leaf.clone();
+        let mut node = proof.entry.compute_leaf();
+
+        let sibling_leaf_node_balances = proof.sibling_leaf_node_hash_preimage[1..]
+            .try_into()
+            .unwrap();
+
+        let sibling_leaf_node_hash = poseidon_entry::<N_ASSETS>(
```

Comment:
poseidon_entry as well as poseidon_node are specific to Node creation and are always used in that context, so their implementations can be placed directly in the Node (no need to have them inside a separate file in utils). For example, here you first create the hash of sibling_leaf_node and then create the sibling_leaf_node Node using this hash, not using the hash for anything else. Meanwhile, Node already has the ::leaf constructor that takes username and balances as args. Moving both hashing operations into the Node would remove confusion over how to create the node.

Reply:
Thanks for the suggestion! Changes applied in #fc44e01. I hope that now it looks cleaner